src/evaluate/evaluator/token_classification.py

```diff
@@ -94,8 +94,19 @@ def words_to_offsets(self, words: List[str], join_by: str):
 
         return offsets
 
-    def prepare_data(self, data: Union[str, Dataset], input_column: str, label_column: str, join_by: str):
-        super().prepare_data(data, input_column, label_column)
+    # def postprocess_data(self, data):
+    #
+    #     return metric_inputs, pipeline_inputs
+    def prepare_data(
+        self, data: Union[str, Dataset], input_column: str, label_column: str, join_by: str, data_split: Optional[str]
+    ):
+        # super().prepare_data(data, input_column, label_column, data_split)
+
+        if isinstance(data, str):
+            data_split = self.get_dataset_split(data, data_split)
+            data = load_dataset(data, split=data_split)
+
+        self.data_with_original_input_column = data[input_column]
```

Comment:
This is symptomatic of something we need to reconcile — line 294 below references the `data` object expecting it to already be a Dataset, which never would've been compatible when passing `data` as a string. Unfortunately the token classification evaluator is the only one which acts like this. It seems that a core issue is that we don't reliably have access to the Dataset object (i.e. before pipeline transformations) outside of `prepare_data` — when `data` is a Dataset that's fine, but when it's a string and needs to be transformed, it's an issue. I'll have to rejig the string-to-Dataset logic a bit everywhere to accommodate.